File: 卡码/代码随想录/Go/二叉树/02-二叉树属性/LeetCode_0257.go
//257. 二叉树的所有路径
package main

import (
	"strconv"
	. "binaryTree/types"
)

func traversal(root *TreeNode,path string,res *[]string){
	path+=strconv.Itoa(root.Val)
	//到达叶子节点
	if root.Left==nil && root.Right==nil {
		(*res)=append((*res),path)
		return
	}
	if root.Left!=nil {
		traversal(root.Left,path+"->",res)
	}
	if root.Right!=nil {
		traversal(root.Right,path+"->",res)
	}
}

func binaryTreePaths(root *TreeNode) []string {
	res:=make([]string,0)
	var path string=""
	if root==nil {
		return res
	}
	traversal(root,path,res)
	return res
}

File: 卡码/代码随想录/Java/codeThoughts/src/main/java/com/kurumi/leetCode/codeThoughts/二叉树/_02_二叉树属性/LeetCode_0257.java
package com.kurumi.leetCode.codeThoughts.二叉树._02_二叉树属性;

import java.util.ArrayList;
import java.util.List;

import com.kurumi.leetCode.codeThoughts.entity.TreeNode;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 257. 二叉树的所有路径
 */
public class LeetCode_0257 {
    static class Solution {
        public void traversal(TreeNode root, String path, List<String> res) {
            path += root.val;
            // 到达叶子节点
            if (root.left == null && root.right == null) {
                res.add(path.toString());
                return;
            }
            if (root.left != null)
                traversal(root.left, path + "->", res);
            if (root.right != null)
                traversal(root.right, path + "->", res);
        }

        public List<String> binaryTreePaths(TreeNode root) {
            List<String> res = new ArrayList<>();
            String path = "";
            if (root == null)
                return res;
            traversal(root, path, res);
            return res;
        }
    }
}
